tests/unit/test_email_function.py

```diff
@@ -1,55 +1,42 @@
 import os
-import unittest
-from mock import patch, Mock
 import sys
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__),'../../src/emailalert')))
-import mock_import
-import email_function
+import unittest
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__),'../../src')))
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__),'../../src/common/python')))
+from emailalert import email_function
+from mock import patch
 
+os.environ["Region"] = "us-west-2"
+os.environ["RecipientEmail"] = "abc@example.com"
+os.environ["SenderEmail"] = "xyz@example.com"
+
+@patch("emailalert.email_function.boto3.client")
 class TestResourceProvider(unittest.TestCase):
-    def test_email_success1(self):
-        os.environ["Region"] = "us-west-2"
-        os.environ["RecipientEmail"] = "abc@example.com"
-        os.environ["SenderEmail"] = "xyz@example.com"
-        factory_patch = patch('email_function.boto3.client')
-        mock_factory_boto_client = factory_patch.start()
-        mock_response = Mock(name='response')
-        mock_factory_boto_client.return_value = mock_response
-        mock_factory_boto_client.send_email.return_value = mock_response
-        mock_response.send_email.return_value = {"MessageId": "Email sent successfully"}
-        event = {"Error": "RestoreError","Cause": "DBInstanceIdentifier:database-1 ThrottlingError: Rate exceeded"}
-        data = email_function.lambda_handler(event, {})
-        self.assertEqual(data.get("Error"), "RestoreError")
-        self.assertEqual(data.get("Cause"), "DBInstanceIdentifier:database-1 ThrottlingError: Rate exceeded")
-        self.assertEqual(data.get("message"), "Email sent successfully")
 
-    def test_email_success2(self):
-        os.environ["Region"] = "us-west-2"
-        os.environ["RecipientEmail"] = "abc@example.com"
-        os.environ["SenderEmail"] = "xyz@example.com"
-        factory_patch = patch('email_function.boto3.client')
-        mock_factory_boto_client = factory_patch.start()
-        mock_response = Mock(name='response')
-        mock_factory_boto_client.return_value = mock_response
-        mock_factory_boto_client.send_email.return_value = mock_response
-        mock_response.send_email.return_value = {"MessageId": "Email sent successfully"}
+    def setUp(self):
+        self.event = {"Error": "RestoreError","Cause": "DBInstanceIdentifier:database-1 ThrottlingError: Rate exceeded"}
+        self.mock_email_response = {"MessageId": "Email sent successfully"}
+
+    def test_email_success1(self, mock_client):
+        mock_ses = mock_client.return_value
+        mock_ses.send_email.return_value = self.mock_email_response
+        data = email_function.lambda_handler(self.event, {})
+        self.assertEqual(data["Error"], "RestoreError")
```

Comment:
If you use assertFalse and assertTrue instead of assertEquals, the output of your unit tests will make more sense when they fail. WDYT ?

Reply:
I think that is basically over engineering the use of assertEqual here as the tests aim to validate the response returned for each function. In case of assert Equals, any failure would directly give an idea on what was the expected vs actual value which i feel is more helpful